src/qibo/hamiltonians/models.py

```diff
@@ -205,11 +124,255 @@ def MaxCut(nqubits, dense: bool = True, backend=None):
+
+    hx = _multikron([matrices.X, matrices.X])
+    hy = _multikron([matrices.Y, matrices.Y])
+    hz = _multikron([matrices.Z, matrices.Z])
+
+    matrix = (
+        -coupling_constants[0] * hx
+        - coupling_constants[1] * hy
+        - coupling_constants[2] * hz
+    )
+
+    terms = [HamiltonianTerm(matrix, i, i + 1) for i in range(nqubits - 1)]
+    terms.append(HamiltonianTerm(matrix, nqubits - 1, 0))
+
+    terms.extend(
+        [
+            -field_strength * HamiltonianTerm(pauli, qubit)
+            for qubit in range(nqubits)
+            for field_strength, pauli in zip(external_field_strengths, paulis)
+            if field_strength != 0.0
+        ]
+    )
+
+    ham = SymbolicHamiltonian(backend=backend)
+    ham.terms = terms
+
+    return ham
+
+
+def XXX(
+    nqubits,
+    coupling_constant: Union[float, int] = 1,
+    external_field_strengths: Union[float, int, list, tuple] = [0.5, 0, 0],
+    dense: bool = True,
+    backend=None,
+):
+    """Heisenberg :math:`\\mathrm{XXX}` model with periodic boundary conditions.
+
+    The :math:`n`-qubit :math:`\\mathrm{XXX}` Hamiltonian is given by
+
+    .. math::
+        H = -J \\, \\sum_{k = 1}^{n} \\, \\left(
+            X_{k} \\, X_{k + 1} + Y_{k} \\, Y_{k + 1} + Z_{k} \\, Z_{k + 1}
+            \\right)
+            - \\sum_{k = 1}^{n} \\left(h_{x} \\, X_{k} + h_{y} \\, Y_{k}
+            + h_{z} \\, Z_{k} \\right) \\, ,
+
+    where :math:`J` is the ``coupling_constant``, :math:`\\{h_{x}, \\, h_{y}, \\, h_{z}\\}`
+    are called the ``external field strengths``, and :math:`\\{X, \\, Y, \\, Z\\}` are the usual
+    Pauli operators.
+
+    Args:
+        nqubits (int): number of qubits.
+        coupling_constant (float or int, optional): coupling constant :math:`J`.
+            Defaults to :math:`1`.
+        external_field_strengths (float or int or list or tuple, optional): list or tuple with the
+            external magnetic field strengths :math:`\\{h_{x}, \\, h_{y}, \\, h_{z}\\}`.
+            If ``int`` or ``float``, then :math:`h_{x} = h_{y} = h_{z}`.
+            Defaults to :math:`[1/2, \\, 0, \\, 0]`.
+        dense (bool, optional): If ``True``, creates the Hamiltonian as a
+            :class:`qibo.core.hamiltonians.Hamiltonian`, otherwise it creates
+            a :class:`qibo.core.hamiltonians.SymbolicHamiltonian`.
+            Defaults to ``True``.
+        backend (:class:`qibo.backends.abstract.Backend`, optional): backend to be used
+            in the execution. If ``None``, it uses the current backend.
+            Defaults to ``None``.
+
+    Returns:
+        :class:`qibo.hamiltonians.Hamiltonian` or :class:`qibo.hamiltonians.SymbolicHamiltonian`:
+        Heisenberg :math:`\\mathrm{XXX}` Hamiltonian.
+    """
+    if not isinstance(coupling_constant, (float, int)):
+        raise_error(
+            TypeError,
+            "`coupling_constant` must be type float or int, "
+            + f"but it is type {type(coupling_constant)}.",
+        )
```

Comment:
Perhaps we could drop this given that a similar check is performed in the `Heiseinberg` function.
https://github.com/qiboteam/qibo/blob/707a106fce7d922fb555b491756fabece6840ec9/src/qibo/hamiltonians/models.py#L183-L187